graylog2-web-interface/src/views/components/searchbar/completions/FieldValueCompletion.ts

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import * as Immutable from 'immutable';
+
+import fetch from 'logic/rest/FetchProvider';
+import { qualifyUrl } from 'util/URLUtils';
+import type { TimeRange, NoTimeRangeOverride } from 'views/logic/queries/Query';
+import { ViewMetadataStore } from 'views/stores/ViewMetadataStore';
+import { FieldTypesStore, FieldTypesStoreState, FieldTypeMappingsList } from 'views/stores/FieldTypesStore';
+import FieldTypeMapping from 'views/logic/fieldtypes/FieldTypeMapping';
+import { onSubmittingTimerange } from 'views/components/TimerangeForForm';
+import { isNoTimeRangeOverride } from 'views/typeGuards/timeRange';
+
+import type { Completer } from '../SearchBarAutocompletions';
+import type { Token } from '../ace-types';
+
+type SuggestionsResponse = {
+  field: string,
+  input: string,
+  suggestions: Array<{ value: string, occurrence: number}> | undefined,
+}
+
+const suggestionsUrl = qualifyUrl('/search/suggest');
+
+const formatValue = (value: string, type: string) => {
+  if (type === 'constant.numeric') {
+    return Number(value);
+  }
+
+  return value;
+};
+
+const getFieldNameAndInput = (currentToken: Token | undefined | null, lastToken: Token | undefined | null) => {
+  if (currentToken?.type === 'keyword' && currentToken?.value.endsWith(':')) {
+    return {
+      fieldName: currentToken.value.slice(0, -1),
+      input: '',
+    };
+  }
+
+  if (currentToken?.type === 'term' && lastToken?.type === 'keyword') {
+    return {
+      fieldName: lastToken.value.slice(0, -1),
+      input: formatValue(currentToken.value, currentToken.type),
+    };
+  }
+
+  return {};
+};
+
+const getFieldByName = (fields: FieldTypeMappingsList, fieldName: string) => {
+  return fields.find(({ name }) => name === fieldName);
+};
+
+const isEnumerableField = (field: FieldTypeMapping) => {
+  return !!field?.type.properties.find((property) => property === 'enumerable');
```

Comment:
We can make this a bit easier:
```suggestion
  return field?.isEnumerable() ?? false;
```